flake-modules/dev/default.nix

```diff
@@ -58,9 +59,22 @@
     }
     // lib.optionalAttrs (inputs.git-hooks ? flakeModule) {
       pre-commit = {
+        # We have a treefmt check already, so this is redundant
+        check.enable = false;
```

Comment:
Hmm, on second thoughts, this made [this error](https://buildbot.nix-community.org/#/builders/120/builds/939) slip through CI in this PR.

Reply:
Looking at the errors on the similar hook added in #2164, I don't think we'll be _able_ to have a pre-commit test if we're running nix commands in it.

The issue is `nix` CLI can't really be used within a derivation builder; and tests run in a derivation builder (since whether it builds or not is the test failure condition).

Therefore, I think we'll need to keep this disablement, expand the comment to mention this issue, and test the hooks work manually...